Source/Blazorise/Components/ToastProvider/ToastProvider.razor

```diff
@@ -13,11 +13,13 @@
                    Animated="@toastInstance.Animated"
                    AnimationDuration="@toastInstance.AnimationDuration"
                    Autohide="@toastInstance.Autohide"
-                   AutohideDelay="@toastInstance.AutohideDelay">
+                   AutohideDelay="@toastInstance.AutohideDelay"
+                   Style="@toastInstance.ToastStyle"
+                   Class="@toastInstance.ToastClass">
                 <ToastHeader>
                     @if ( ShowIntentIcon )
                     {
-                        <Icon Name="@GetIconName( toastInstance.Intent )" TextColor="@GetIconColor( toastInstance.Intent )" Margin="Blazorise.Margin.Is2.FromEnd" />
+                        <Icon Name="@(toastInstance.IconName ?? GetIconName( toastInstance.Intent ))" TextColor="@GetIconColor( toastInstance.Intent )" Margin="Blazorise.Margin.Is2.FromEnd" />
```

Comment:
Pass the `toastInstance` to the `GetIconName` and have it decide what to report based on the instance state.